packages/tpetra/core/test/MultiVector/MultiVector_MixedNode.cpp

```diff
@@ -0,0 +1,263 @@
+#if ! defined(TRILINOS_UNUSED_FUNCTION)
+#  if defined(__GNUC__) || (defined(__INTEL_COMPILER) && !defined(_MSC_VER))
+#    define TRILINOS_UNUSED_FUNCTION __attribute__((__unused__))
+#  elif defined(__clang__)
+#    if __has_attribute(unused)
+#      define TRILINOS_UNUSED_FUNCTION __attribute__((__unused__))
+#    else
+#      define TRILINOS_UNUSED_FUNCTION
+#    endif // Clang has 'unused' attribute
+#  elif defined(__IBMCPP__)
+// IBM's C++ compiler for Blue Gene/Q (V12.1) implements 'used' but not 'unused'.
+//
+// http://pic.dhe.ibm.com/infocenter/compbg/v121v141/index.jsp
+#    define TRILINOS_UNUSED_FUNCTION
+#  else // some other compiler
+#    define TRILINOS_UNUSED_FUNCTION
+#  endif
+#endif // ! defined(TRILINOS_UNUSED_FUNCTION)
+
+
+namespace Teuchos {
+  template <>
+  ScalarTraits<int>::magnitudeType
+  relErr( const int &s1, const int &s2 ) {
+    typedef ScalarTraits<int> ST;
+    return ST::magnitude(s1-s2);
+  }
+
+  template <>
+  ScalarTraits<char>::magnitudeType
+  relErr( const char &s1, const char &s2 ) {
+    typedef ScalarTraits<char> ST;
+    return ST::magnitude(s1-s2);
+  }
+}
+
+namespace {
+
+  using Tpetra::TestingUtilities::getDefaultComm;
+
+  using std::endl;
+  using std::copy;
+  using std::string;
+
+  using Teuchos::Array;
+  using Teuchos::ArrayRCP;
+  using Teuchos::ArrayView;
+  using Teuchos::arrayView;
+  using Teuchos::as;
+  using Teuchos::Comm;
+  using Teuchos::null;
+  using Teuchos::Range1D;
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+  using Teuchos::REDUCE_MIN;
+  using Teuchos::reduceAll;
+  using Teuchos::OrdinalTraits;
+  using Teuchos::outArg;
+  using Teuchos::ScalarTraits;
+  using Teuchos::SerialDenseMatrix;
+  using Teuchos::Tuple;
+  using Teuchos::tuple;
+  using Teuchos::NO_TRANS;
+  using Teuchos::TRANS;
+  using Teuchos::CONJ_TRANS;
+  using Teuchos::VERB_DEFAULT;
+  using Teuchos::VERB_NONE;
+  using Teuchos::VERB_LOW;
+  using Teuchos::VERB_MEDIUM;
+  using Teuchos::VERB_HIGH;
+  using Teuchos::VERB_EXTREME;
+
+  using Tpetra::Map;
+  using Tpetra::MultiVector;
+  using Tpetra::global_size_t;
+  using Tpetra::GloballyDistributed;
+
+  using Tpetra::createContigMapWithNode;
+  using Tpetra::createLocalMapWithNode;
+
+  double errorTolSlack = 1.0e+2;
+
+  TEUCHOS_STATIC_SETUP()
+  {
+    Teuchos::CommandLineProcessor &clp = Teuchos::UnitTestRepository::getCLP();
+    clp.addOutputSetupOptions(true);
+    clp.setOption(
+        "test-mpi", "test-serial", &Tpetra::TestingUtilities::testMpi,
+        "Test MPI (if available) or force test of serial.  In a serial build,"
+        " this option is ignored and a serial comm is always used." );
+    clp.setOption(
+        "error-tol-slack", &errorTolSlack,
+        "Slack off of machine epsilon used to check test results" );
+  }
+
+  // no ScalarTraits<>::eps() for integer types
+  template <class Scalar>
+  typename Teuchos::ScalarTraits<Scalar>::magnitudeType testingTol() { return Teuchos::ScalarTraits<Scalar>::eps(); }
```

Comment:
You don't use these in the file, so they can go away.  That also makes the `TRILINOS_UNUSED_FUNCTION` logic unnecessary.